skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+
+        The ``value`` should be ``None`` (meaning all boundaries are ``True``)
+        or an iterable matching ``self.locations``.'''
+        if self.dropped:
+            raise RuntimeError('Cannot change boundaries to dropped '
+                               'Interval object.')
+        nl = len(self.locations)
+        if value is None:
+            self._boundaries = [(True, True)] * nl
+        else:
+            try:
+                value = list(value)
+            except:
+                raise TypeError(
+                    'Cannot give a non-iterable (%r) to `boundaries`.' % value)
+
+            nb = len(value)
+            if nb != nl:
+                msg = ('The length of boundaries (%d) needs '
+                       'to be equal to the length of locations (%d).')
+                raise ValueError(msg % (nb, nl))
+
+            for boundary in value:
+                _assert_valid_boundary(boundary)
+            self._boundaries = value
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def locations(self):
+        '''The coordinates of the interval feature.
+
+        It should be a list of tuples of int pair. Each tuple stores
+        the start and end coordinates of a span of the interval feature.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve locations from dropped Interval object.')
+        return self._locations
+
+    @locations.setter
+    @experimental(as_of='0.5.0-dev')
+    def locations(self, value):
+        '''Set ``locations``.
+
+        WARNING: setting ``locations`` will reset ``boundaries`` value to True.
+        This is not totally surprising because it is justifiable your old
+        boundaries don't fit the new locations.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot change locations to dropped Interval object.')
+        # check iterability
+        try:
+            value = list(value)
+        except:
+            raise TypeError(
+                'Cannot give a non-iterable (%r) to `locations`.' % value)
+        # check it is not empty
+        if len(value) == 0:
+            raise ValueError(
+                'Cannot give empty `location`.')
+        # check each contiguous span is in right format
+        for location in value:
+            _assert_valid_location(location)
+        # don't forget to sort the spans
+        self._locations = sorted(value)
+        # reset all the boundaries to True!!
+        self.boundaries = None
+        self._interval_metadata._is_stale_tree = True
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def metadata(self):
+        '''The metadata of the interval feature.
+
+        It stores the metadata (eg. gene name, function, ID, etc.) of
+        the interval feature as a ``dict``.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve metadata from dropped Interval object.')
```

Comment:
I think we do want to be able to retrieve the metadata still, it shouldn't be dependent on the presence/absence of the IntervalMetadata object.